Mathlib/Analysis/Normed/ValuativeRel.lean

```diff
@@ -0,0 +1,101 @@
+/-
+Copyright (c) 2025 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+import Mathlib.Analysis.Normed.Field.Lemmas
+import Mathlib.Analysis.Normed.Group.Ultra
+import Mathlib.RingTheory.Valuation.RankOne
+import Mathlib.Topology.Algebra.Valued.ValuativeRel
+
+/-!
+
+# Nonarchimedean normed rings carry a valuative topology
+
+In this file, we provide a helper instance
+for `ValuativeRel R` derived from a `NormedRing R` when `NormMulClass R` and `IsUltrametricDist R`,
+so that downstream files can refer to `ValuativeRel R`,
+to facilitate a refactor.
+
+## Implementation details
+
+The namespace is `NormedField` because we need `‖x * y‖ = ‖x‖ * ‖y‖`, not just submultiplicativity.
+The `NormedCommRing` assumption is not enough because that gives `‖x * y‖ ≤ ‖x‖ * ‖y‖`.
+The associated prop classes of `NormOneClass` and `NormMulClass` give the full multiplicativity.
+
+-/
+
+variable {R : Type*} [NormedCommRing R] [NormOneClass R]
+  [NormMulClass R] [IsUltrametricDist R]
+
+open Filter Set Valuation
+
+open scoped NNReal
+
+namespace NormedField
+
+/-- The valuation on a nonarchimedean normed field `K` defined as `nnnorm`.
+Should replace `NormedField.valuation` in the future. -/
+def valuation' : Valuation R ℝ≥0 where
+  toFun           := nnnorm
+  map_zero'       := nnnorm_zero
+  map_one'        := nnnorm_one
+  map_mul'        := nnnorm_mul
+  map_add_le_max' := IsUltrametricDist.norm_add_le_max
+
+@[simp]
+theorem valuation'_apply (x : R) : valuation' x = ‖x‖₊ := rfl
+
+/-- A nonarchimedean normed ring carries a valuative relation induced by the norm.
+This is a scoped instance. -/
+def toValuativeRel : ValuativeRel R := .ofValuation valuation'
```

Comment:
```suggestion
@[reducible] def toValuativeRel : ValuativeRel R := .ofValuation valuation'
```
See `library note[reducible non-instance]`

Reply:
Why? The only thing that is mentioned in the instance is other `R` instances. I will change it though (`abbrev`), just trying to learn.

Reply:
My understanding is that `instance`s are automatically reducible so scoped instances should get tagged reducible for consistency and predictability. See the library note somewhere in mathlib for a better/more detailed argument.